app/routes/goal_routes.py

```diff
@@ -0,0 +1,96 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, request, jsonify, make_response
+from app.routes.task_routes import validate_id
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# ================================
+# Create one goal 
+# ================================
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except KeyError:
+        return make_response({"details": "Invalid data"}, 400)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return jsonify({"goal": new_goal.to_json()}), 201
+
+# ==================================
+# Get all goals  
+# ==================================
+
+@goal_bp.route("", methods=["GET"])
+def get_all_goals():
+        all_goals = Goal.query.all()
+        results_list = []
+        for goal in all_goals:
+            results_list.append(goal.to_json())
+        return jsonify(results_list), 200
+
+# ==================================
+# Get one goal by id number
+# ==================================
+
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    return jsonify({"goal": validate_id(Goal, goal_id).to_json()}), 200
```

Comment:
This works fine and is succinct, but there is a lot going on that may make this a bit hard to parse for a reader. That is, there's a call to `validate_id`, `to_json`, the dictionary constructor, _and_ `jsonify`. (Which, you don't need the `jsonify` call as previously mentioned.)

Beyond just being hard to parse, this also has disadvantages when debugging. For example, if you wanted to debug _just_ this call to `to_json` by putting a breakpoint on it,  you'd have to step in and out of `validate_id` before that calls `to_json`. Alternatively, you put your breakpoint in `to_json`, but then _all_ calls to `to_json` would hit that breakpoint unless you make a conditional breakpoint (which I'm not sure there's an easy condition you could use here...).

I'm probably being nitpicky, but I guess I wanted to point out some of the issues with succinctness!